mhr_api/src/mhr_api/services/document_storage/storage_service.py

```diff
@@ -0,0 +1,211 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This class is a wrapper for document storage API calls."""
+import os
+import json
+import urllib.parse
+from abc import ABC, abstractmethod
+from enum import Enum
+
+import requests
+from flask import current_app
+from google.cloud import storage
+
+from mhr_api.services.gcp_auth.token_service import GoogleStorageTokenService
+from mhr_api.services.utils.exceptions import StorageException
+
+
+HTTP_DELETE = 'delete'
+HTTP_GET = 'get'
+HTTP_POST = 'post'
+
+
+class DocumentTypes(str, Enum):
+    """Render an Enum of storage document types."""
+
+    SEARCH_RESULTS = 'SEARCH_RESULTS'
+    REGISTRATION = 'REGISTRATION'
+
+
+class StorageService(ABC):  # pylint: disable=too-few-public-methods
```

Comment:
In mine I separated out the abstract class so that the "interface" could be imported without dragging along the implementation specific libraries.

Reply:
Separated out the abstract class.